### PR TITLE
[WIP] Add go integration tests

### PR DESCRIPTION
**Pull request by jcstorms1**
This PR adds 3 new integration tests for the Go runtime in preparation for upcoming changes to the Lambda Extension.

- a metric test with the datadog lambda library
- a metric test without the datadog lambda library
- a timeout test
- a log test
- a trace test

### CODE REVIEW

File: integration_tests/run.sh [hunk 1]
```diff
@@ -36,47 +36,51 @@ cd "./integration_tests"
...
 
 function remove_stack() {
     echo "Removing stack for stage : ${stage}"
-    serverless remove --stage ${stage} 
+    serverless remove --stage ${stage}
 }
 
 # always remove the stacks before exiting, no matter what
-trap remove_stack EXIT
+# trap remove_stack EXIT
```
Comment (nhinsch) [inline integration_tests/run.sh:74]: I think we still want this line uncommented

File: integration_tests/src/go-tests/without-ddlambda/main.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+	"github.com/aws/aws-lambda-go/events"
+	"github.com/aws/aws-lambda-go/lambda"
+)
+
+type testResponse struct {
+	StatusCode int    `json:"statusCode"`
+	Body       string `json:"body"`
+}
+
+func PingHandler(ctx context.Context, ev events.APIGatewayProxyRequest) (testResponse, error) {
```
Comment (nhinsch) [inline integration_tests/src/go-tests/without-ddlambda/main.go:15]: Let's add a log line here to make sure logging works without the ddlambda library

Comment (nhinsch) [reply integration_tests/src/go-tests/without-ddlambda/main.go:15]: Actually, I think you could remove the "logs test" and move the log lines into the with-dd-lambda and without-dd-lambda tests. That way we cover logs in both cases.

File: integration_tests/run.sh (hunk 1 above)
Comment (jcstorms1) [reply integration_tests/run.sh:74]: Ahh yeah forgot I did that. Thanks!

File: integration_tests/serverless.yml [hunk 3]
```diff
@@ -107,6 +126,16 @@ functions:
...
 
+  timeout-go:
+    runtime: provided.al2
+    timeout: 3
+    handler: src/bin/timeout
+    layers:
+      - { Ref: RecorderExtensionLambdaLayer }
+      - { Ref: DatadogExtensionIntegrationTestLambdaLayer }
+    environment:
+      DD_LOG_LEVEL: DEBUG
```
Comment (nhinsch) [inline integration_tests/serverless.yml:137]: Why do you want to set the log level to `DEBUG` for this test?

File: integration_tests/serverless.yml [hunk 4]
```diff
@@ -86,6 +96,15 @@ functions:
       DD_ENHANCED_METRICS: false
       DD_LAMBDA_HANDLER: src/metricPython.metric
 
+  no-enhanced-metric-go:
```
Comment (nhinsch) [inline integration_tests/serverless.yml:99]: This function is called "no-enhanced-metric-go" but the handler is testing without the Lambda Layer. I think what we are really testing here is without the Lambda layer, so let's name the function based on that.

Comment (jcstorms1) [reply integration_tests/serverless.yml:99]: 👍

File: integration_tests/serverless.yml (hunk 3 above)
Comment (jcstorms1) [reply integration_tests/serverless.yml:137]: I just followed the other tests. Not sure that it matters.

File: integration_tests/run.sh [hunk 5]
```diff
@@ -36,50 +36,58 @@ cd "./integration_tests"
...
 cd recorder-extension
-    GOOS=linux GOARCH=amd64 go build -o extensions/recorder-extension main.go
-    zip -rq ext.zip extensions/* -x ".*" -x "__MACOSX" -x "extensions/.*"
+GOOS=linux GOARCH=amd64 go build -o extensions/recorder-extension main.go
+zip -rq ext.zip extensions/* -x ".*" -x "__MACOSX" -x "extensions/.*"
 cd ..
 
+go_test_dirs=("with-ddlambda" "without-ddlambda" "timeout" "trace")
+
 # build Go Lambda function
```
Comment (nhinsch) [inline integration_tests/run.sh:45]: Nit: function -> functions

File: integration_tests/snapshots/log-go [hunk 6]
```diff
@@ -0,0 +1,10 @@
+[log] {"message":{"message":"START XXX"}}
```
Comment (nhinsch) [inline integration_tests/snapshots/log-go:1]: What function is this snapshot (`log-go`) for? Is this from an older function that you deleted?

**Review by nhinsch — APPROVED** [review]
Looks good, great work! Just left a couple small notes.

File: integration_tests/serverless.yml [hunk 7]
```diff
@@ -86,6 +100,21 @@ functions:
...
       DD_LAMBDA_HANDLER: src/metricPython.metric
 
+  without-ddlambda-go:
+    runtime: provided.al2
+    handler: src/bin/without-ddlambda
+    layers:
+      - { Ref: RecorderExtensionLambdaLayer }
+      - { Ref: DatadogExtensionIntegrationTestLambdaLayer }
+    environment:
+      DD_ENHANCED_METRICS: false
```
Comment (nhinsch) [inline integration_tests/serverless.yml:110]: I would set this to true and regenerate the snapshots.

**Pull request merged**